pkg/cache/snapshot.go

```diff
@@ -170,7 +169,7 @@ func snapshotClusterQueue(c *clusterQueue) *ClusterQueueSnapshot {
 		Preemption:                    c.Preemption,
 		NamespaceSelector:             c.NamespaceSelector,
 		Status:                        c.Status,
-		AdmissionChecks:               utilmaps.DeepCopySets[kueue.ResourceFlavorReference](c.AdmissionChecks),
```

Comment:
please delete utilmaps.DeepCopySets (assuming no other usages)

Reply:
Actually, this is changing the semantics, no? While we are cloning the maps, they are still referencing the same memory for admission check sets. Please revert

Reply:
Oh, I see what you mean @gabesaba , thanks for pointing it out.  If we don't clone the AC sets (values) the CQ snapshot will "see" the changes done to the underlying CQ. Snapshot needs to be "frozen" to allow correct scheduling decisions. This could result in very subtle bugs.

I would suggest to continue using `utilmaps.DeepCopySets`, but parametrize the type for keys which is currently hardcoded as string.